src/projections/genericsom.cpp

```diff
@@ -173,29 +186,11 @@ static PJ_LP misrsom_e_inverse (PJ_XY xy, PJ *P) {          /* Ellipsoidal, inve
     return lp;
 }
 
-
-PJ *PROJECTION(misrsom) {
-    int path;
-    double lam, alf, esc, ess;
-
-    struct pj_opaque *Q = static_cast<struct pj_opaque*>(calloc (1, sizeof (struct pj_opaque)));
-    if (nullptr==Q)
-        return pj_default_destructor (P, PROJ_ERR_OTHER /*ENOMEM*/);
-    P->opaque = Q;
-
-    path = pj_param(P->ctx, P->params, "ipath").i;
-    if (path <= 0 || path > 233)
-    {
-        proj_log_error(P, _("Invalid value for path: path should be in [1, 233] range"));
-        return pj_default_destructor(P, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
-    }
-
-    P->lam0 = DEG_TO_RAD * 129.3056 - M_TWOPI / 233. * path;
-    alf = 98.30382 * DEG_TO_RAD;
-    Q->p22 = 98.88 / 1440.0;
-
-    Q->sa = sin(alf);
-    Q->ca = cos(alf);
+static PJ *setup(PJ *P, SOMProjectionType somProjType) {
```

Comment:
this somProjType parameter isn't use, so it could be removed and the SOMProjectionType enumeration itself too

Reply:
I'm currently trying to unify with lsat.cpp as well, where this will be in use. (Assuming that is okay)

Reply:
> I'm currently trying to unify with lsat.cpp as well, where this will be in use. (Assuming that is okay)

That would be fantastic! Indeed the side-to-side comparison between lsat.cpp and misrom.cpp shows a lot of similarities !

Reply:
I moved init earlier so that this parameter was no longer required (and neither was the enum).